prerequisites/install-functions/build_opencoarrays.sh

```diff
@@ -10,18 +10,27 @@ build_opencoarrays()
   build_path="${build_path}"/opencoarrays/$("${opencoarrays_src_dir}"/install.sh -V opencoarrays)
   mkdir -p "$build_path"
   pushd "$build_path"
-  if [[ -z "${MPICC:-}" || -z "${MPIFC:-}" || -z "${CMAKE:-}" || -z "${FC:-}" || -z "${CC:-}" ]]; then
-    emergency "Empty MPICC=$MPICC or MPIFC=$MPIFC or FC or CC or CMAKE=$CMAKE [exit 90]"
-  else
-    info "Configuring OpenCoarrays in ${PWD} with the command:"
-    info "CC=\"${CC}\" FC=\"${FC}\" $CMAKE \"${opencoarrays_src_dir}\" -DCMAKE_INSTALL_PREFIX=\"${install_path}\" -DMPI_C_COMPILER=\"${MPICC}\" -DMPI_Fortran_COMPILER=\"${MPIFC}\""
-    CC="${CC}" FC="${FC}" $CMAKE "${opencoarrays_src_dir}" -DCMAKE_INSTALL_PREFIX="${install_path}" -DMPI_C_COMPILER="${MPICC}" -DMPI_Fortran_COMPILER="${MPIFC}"
-    info "Building OpenCoarrays in ${PWD} with the command make -j${num_threads}"
-    make "-j${num_threads}"
-    if [[ ! -z ${SUDO:-} ]]; then
-      printf "\nThe chosen installation path requires sudo privileges. Please enter password if prompted.\n"
-    fi
-    info "Installing OpenCoarrays in ${install_path} with the command ${SUDO:-} make install"
-    ${SUDO:-} make install
+  if [[ -z ${MPIFC:-} || -z ${MPICC:-} ]]; then
+    emergency "build_opencoarrays.sh: empty \${MPIFC}=${MPIFC:-} or \${MPICC}=${MPICC:-}"
   fi
+  MPIFC_show=`$MPIFC -show`
+  MPICC_show=`$MPICC -show`
+  if [[ ${MPIFC_show} != *gfortran* || ${MPICC_show} != *gcc* ]]; then
+    emergency "build_opencoarrays.sh: MPI doesn't wrap gfortran/gcc: \${MPIFC_show}=${MPIFC_show}, \${MPICC_show}=${MPICC_show}"
+  fi 
+  # Set FC to the MPI implementation's gfortran command with any preceding path but without any subsequent arguments:
+  FC="${MPIFC_show%%gfortran*}"gfortran
+  # Set CC to the MPI implementation's gcc command...
+  CC="${MPICC_show%%gcc*}"gcc
+  info "Configuring OpenCoarrays in ${PWD} with the command:"
+  info "CC=\"${CC}\" FC=\"${FC}\" $CMAKE \"${opencoarrays_src_dir}\" -DCMAKE_INSTALL_PREFIX=\"${install_path}\" -DMPI_C_COMPILER=\"${MPICC}\" -DMPI_Fortran_COMPILER=\"${MPIFC}\""
+  CC="${CC}" FC="${FC}" $CMAKE "${opencoarrays_src_dir}" -DCMAKE_INSTALL_PREFIX="${install_path}" -DMPI_C_COMPILER="${MPICC}" -DMPI_Fortran_COMPILER="${MPIFC}"
+  info "Building OpenCoarrays in ${PWD} with the command make -j${num_threads}"
+  make "-j${num_threads}"
+  if [[ ! -z ${SUDO:-} ]]; then
+    printf "\nThe chosen installation path requires sudo privileges. Please enter password if prompted.\n"
+  fi
+  info "Installing OpenCoarrays in ${install_path} with the command ${SUDO:-} make install"
+  ${SUDO:-} make install
```

Comment:
LGTM 

Thanks for the quick feedback.  I think it's ready.  I tested the following scenarios:
- In the VM with -f/-c/-C arguments pointing to GCC 7 dated 20161112.  
- In MacOS with GCC 6.2.0 
    - with -f/-c/-C arguments.
    - with no arguments
- In MacOS with `FC=pgortran CC=pgcc ./install.sh` just to see if it went for the head fake ⛹️, which it did not. 😉 

All 29 tests pass in MacOS.  In the VM, 29 of 30 tests pass with the one previously observed failure being the get_self test.  I'm going o launch a new build of the current trunk in the VM tonight to see if the one test failure remains, but that's unrelated to this pull request so merge away!  (assuming the Travis-CI tests pass)

I also tried